platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/conflicts/DefaultCapabilitiesConflictHandler.java

```diff
@@ -116,6 +117,21 @@ public boolean hasConflicts() {
     @Override
     public void resolveNextConflict(Action<ConflictResolutionResult> resolutionAction) {
         CapabilityConflict conflict = conflicts.poll();
+
+        List<NodeState> selectedConflictingNodes =
+            conflict.nodes.stream().filter(NodeState::isSelected).collect(Collectors.toList());
+
+        if (selectedConflictingNodes.isEmpty()) {
+            return;
+        } else if (selectedConflictingNodes.size() < conflict.nodes.size()) {
+            // We remove any nodes that are not selected from the conflict-to-resolve.
+            conflict = new CapabilityConflict(
+                conflict.group,
+                conflict.name,
+                selectedConflictingNodes
+            );
+        }
```

Comment:
💅  The logic above seems like it could be simplified.  Wouldn't the following achieve the same:
```
conflict = new CapabilityConflict(
                conflict.group,
                conflict.name,
                conflict.nodes.stream().filter(NodeState::isSelected).collect(Collectors.toList());
            );

if (conflict.nodes.isEmpty()) {
    return;
}
```

You could even pull the filtering into a `CapabilityConflict.withSelectedConflictingNodes()` method, so then it just becomes:
```
CapabilityConflict conflict = conflicts.poll().withSelectedConflictingNodes();

if (conflict.nodes.isEmpty()) {
    return;
}
```